### PR TITLE
sql, README: improve JSON doc

### PR DESCRIPTION
**Pull request by CaitinChen**
Via https://github.com/pingcap/docs/pull/656
@lilin90 @zz-jason PTAL

### CODE REVIEW

**Comment by morgo** [comment]
Note: one of the 'unsupported' functions has since been merged:  https://github.com/pingcap/docs/pull/680

(Up to you if you want to handle in this PR, or a followup one.)

File: sql/generated-columns.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+CREATE TABLE person (
+    id INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    name VARCHAR(255) NOT NULL,
+    address_info JSON,
+    city VARCHAR(64) AS (JSON_UNQUOTE(JSON_EXTRACT(address_info, '$.city'))) VIRTUAL,
+    KEY (city)
+);
+```
+
+该表中，`city` 列是一个 **generated column**。顾名思义，此列由该表的其他列生成，对此列进行插入或更新操作时，并不能对之赋值。此列按需生成，并不存储在数据库中，也不占用内存空间，因而是_虚拟的_。`city` 列的索引_存储在数据库中_，并使用和 `varchar(64)` 类的其他索引相同的结构。
```
Comment (lilin90) [inline sql/generated-columns.md]: The italic effect does not work well here. You can click "View" to check it. 

In Chinese text, limit the use of the italic effect. For emphasis, the bold effect is preferred.

File: sql/generated-columns.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ERROR 1048 (23000): Column 'city' cannot be null
+```
+
+## 局限性
+
+目前 JSON and generated column 有以下局限性：
+
+- 不能通过 `ALTER TABLE` 增加 `STORED` 存储方式的 generated column；
+- 不能通过 `ALTER TABLE` 在 generated column 上增加索引；
+- 并未支持所有的 [JSON 函数](json-functions.md)。
```
Comment (lilin90) [inline sql/generated-columns.md]: Please update the link path to `../sql/json-functions.md`.

File: sql/json-functions.md [hunk 3]
```diff
@@ -1,26 +1,68 @@
 ---
-title: JSON 相关的函数和语法糖
+title: JSON 函数及语法糖
 category: user guide
 ---
 
-# JSON 相关的函数和语法糖
+# JSON 相关的函数及语法糖
```
Comment (lilin90) [inline sql/json-functions.md]: Keep this heading consistent with the title in metadata.

File: sql/json-functions.md [hunk 4]
```diff
@@ -1,26 +1,68 @@
...
+| [JSON_TYPE(json_val)][json_type] | 检查某 JSON 文档内部内容的类型 |
+
+## 未支持的函数
+ 
+TiDB 暂未支持以下 JSON 函数。相关进展参见 [TiDB #7546](https://github.com/pingcap/tidb/issues/7546):
+
+* `JSON_APPEND` 及其别名 `JSON_ARRAY_APPEND`
+* `JSON_ARRAY_INSERT`
+* `JSON_DEPTH`
+* `JSON_KEYS`
```
Comment (lilin90) [inline sql/json-functions.md]: As what Morgan has mentioned, this function has been removed from the unsupported list. See https://github.com/pingcap/docs/pull/680

File: sql/json-functions.md [hunk 5]
```diff
@@ -30,3 +72,10 @@ category: user guide
 [json_merge]: https://dev.mysql.com/doc/refman/5.7/en/json-modification-functions.html#function_json-merge
 [json_object]: https://dev.mysql.com/doc/refman/5.7/en/json-creation-functions.html#function_json-object
 [json_array]: https://dev.mysql.com/doc/refman/5.7/en/json-creation-functions.html#function_json-array
+[json_keys]: https://dev.mysql.com/doc/refman/5.7/en/json-search-functions.html#function_json-keys
```
Comment (lilin90) [inline sql/json-functions.md]: Please also check and update the links. If it links to no place, please remove it.

File: sql/generated-columns.md (hunk 1 above)
Comment (zz-jason) [reply sql/generated-columns.md]: How about "因而是**虚拟的**" Or "因而是 _虚拟的_ 。"?
```
因而是**虚拟的**
因而是 _虚拟的_ 。 // NOTE: There is a space to separate the _ symbol.
```

Comment (CaitinChen) [reply sql/generated-columns.md]: Thank you both. Generally, we do not use the italic effect in Chinese text, so I will change it to the bold style.

**Pull request merged**